libzstd/src/lib.rs

```diff
@@ -29,7 +29,21 @@ pub unsafe extern "C" fn compress_scroll_batch_bytes(
     let src = unsafe { slice::from_raw_parts(src, src_size as usize) };
     let out = unsafe { slice::from_raw_parts_mut(output_buf, buf_size as usize) };
 
-    let mut encoder = init_zstd_encoder(N_BLOCK_SIZE_TARGET);
+    // when the src size larger than one input block (so the output has too
+    // be distributed in multiple blocks), we average each input block to
+    // avoid the small size in last input, which may lead to Raw or RLE blocks
+    let mut encoder = if src_size > N_BLOCK_SIZE_TARGET as u64{
+        let exp_blocks = (src_size - 1) / N_BLOCK_SIZE_TARGET as u64 + 1;
+        /* 
+            slightly increase for blocks except the last one,
+            so ensure no "tip" would be left
+        */
+        let reset_blk_size = src_size / exp_blocks + 8;
```

Comment:
Is this safe?

so basically what we had is 

(`src_size` - 1) / (`exp_blocks` - 1 ) = `N_BLOCK_SIZE_TARGET`


now we want to pick a `reset_blk_size` so that 


(`src_size` - 1) / (`reset_exp_blocks` - 1 ) != reset_blk_size


but if `reset_blk_size = src_size / exp_blocks + 8`

as you can see, 
this equation is still possible

![QianJianTec1720365273853](https://github.com/scroll-tech/da-codec/assets/37070449/e0d86d9e-5dd9-49ba-b701-2e0db4f4e326)

Reply:
Could you explain where the `reset_exp_blocks` comes from?

Reply:
> Could you explain where the `reset_exp_blocks` comes from?

similar to `exp_blocks` = `src_size` / `N_BLOCK_SIZE_TARGET`,
 `reset_exp_blocks` is `src_size` / `reset_blk_size`